### ISSUE TITLE
`super` should allow calling methods from merged interface

### ISSUE THREAD

**Issue by vclav** [Microsoft/TypeScript#13563]
**TypeScript Version:**  2.1.5

SUGGESTION: It seems to be no way to define extending for a `class`

**Code**
```ts

class Classic {
    
}

// some global object literal containing methods
declare window.somePOJO: { [key: string]: Function };
// dirty inheritance of POJO
Object.assign(Classic.prototype, window.somePOJO);
```

**Question:** How tot tell TS, that Classic is extended (has new methods)?

**Attempt 1 -- Interface**

```ts
class Classic ...
object.Assign(...

// declare inherited methods in interface of Classic
interface Classic {
    inheritedMethod(): void;
}

// but then in subClass
class SubClassic extends Classic {
    inheritedMethod() {
        super.inheritedMethod(); // Error TS2340´
        // Only public and protected methods of the base class are accessible via the 'super' 
    }
}
```

**Attempt 2 -- Declare POJO as base class**

```ts

// all methods of POJO in declared class
declare class somePOJO {
    public inheritedMethod(): void;
}

class Classic extends somePOJO {
    constructor() {
        // Error TS17009
        // 'super' must be called before accessing 'this' in the constructor of a derived class.
        // But super on POJO fails
    }
}
```

I searched hard in documentation and asked on [StackOverflow](http://stackoverflow.com/q/41708394/2508019), but defining extensions for class seems to be currently unsupported.

**Comment by RyanCavanaugh** [comment]
In the absence of knowing either way, I think we should allow `super` calls of method-syntax-declared properties in a class instance augmentation (as shown at `super.inheritedMethod()`). @mhegazy thoughts?

**Comment by mhegazy** [comment]
seems like a bug indeed.

**Comment by sandersn** [comment]
Fix is up at #13726
